src/Symfony/Component/DependencyInjection/Compiler/CheckDefinitionValidityPass.php

```diff
@@ -48,6 +48,14 @@ public function process(ContainerBuilder $container)
                 if ($definition->getFactory()) {
                     throw new RuntimeException(sprintf('Please add the class to service "%s" even if it is constructed by a factory since we might need to add method calls based on compile-time checks.', $id));
                 }
+                if (class_exists($id) || interface_exists($id)) {
+                    throw new RuntimeException(sprintf(
+                         'Service "%s" appears to reference a class in the global namespace, and does '
+                        .'not have a class defined. Leaving out the class property is only allowed '
+                        .'for namespaced classes. Specify the class explicitly to get rid of this error.',
```

Comment:
That is imho less clear as the "appears to reference a class in the global namespace" part is key to the DX - it explains that what the developer is trying to do is not allowed, and then why. In your suggestion it's not emphasized that the global namespace is the issue.

I do use the word property instead of attribute. Is that convention?